scripts/cnv_wdl/cnv_common_tasks.wdl

```diff
@@ -292,16 +294,35 @@ task ScatterIntervals {
     Int? preemptible_attempts
 
     Int machine_mem_mb = select_first([mem_gb, 2]) * 1000
+    Int command_mem_mb = machine_mem_mb - 500
+
+    # If optional output_dir not specified, use "out";
+    String output_dir_ = select_first([output_dir, "out"])
 
     String base_filename = basename(interval_list, ".interval_list")
 
     command <<<
         set -e
-
-        grep @ ${interval_list} > header.txt
-        grep -v @ ${interval_list} > all_intervals.txt
-        split -l ${num_intervals_per_scatter} --numeric-suffixes all_intervals.txt ${base_filename}.scattered.
-        for i in ${base_filename}.scattered.*; do cat header.txt $i > $i.interval_list; done
+        mkdir ${output_dir_}
+        export GATK_LOCAL_JAR=${default="/root/gatk.jar" gatk4_jar_override}
+        
+        {
+            >&2 echo "Attempting to run IntervalListTools..."
+            gatk --java-options "-Xmx${command_mem_mb}m" IntervalListTools \
+                --INPUT ${interval_list} \
+                --SUBDIVISION_MODE INTERVAL_COUNT \
+                --SCATTER_CONTENT ${num_intervals_per_scatter} \
+                --OUTPUT ${output_dir_} &&
+            # output files are named output_dir_/temp_0001_of_N/scattered.interval_list, etc. (N = num_intervals_per_scatter);
+            # we rename them as output_dir_/base_filename.scattered.0000.interval_list, etc.
+            ls ${output_dir_}/*/scattered.interval_list | \
+                cat -n | \
+                while read n filename; do mv $filename ${output_dir_}/${base_filename}.scattered.$(printf "%04d" $n).interval_list; done
+        } || {
+            # if only a single shard is required, then we can just rename the original interval list
```

Comment:
Is it possible to handle this without waiting for IntervalListTools to fail?

Reply:
Also, this obviates the need to duplicate the logic for calculating SCATTER_COUNT in IntervalListTools (specifically, the decision for what to do when the last scatter block is smaller than the block size), which is desired